### ISSUE TITLE
Creating and calling 2 models one after another

### ISSUE THREAD

**Issue by aidonchuk** [NVIDIA/retinanet-examples#124]
Hi, have a problem crashing inference when 2 models created and executed one after another
If i comment model1 inference every think goes ok. Models exported by cppapi

import random
import time

import torch
import torch.nn.functional as F
from PIL import Image

from retinanet._C import Engine

def get_model_trt(path):
return Engine.load(path)

def infer_trt(im_path, resize, max_size):
model = get_model_trt(path='../models/retinanet_rn18fpn_1_stage.engine')
model1 = get_model_trt(path='../models/retinanet_rn34fpn_2_stage.engine')

im = Image.open(im_path).convert("RGB")

if isinstance(resize, list):
resize = random.randint(resize[0], resize[-1])

ratio = resize / min(im.size)
if ratio * max(im.size) > max_size:
ratio = max_size / max(im.size)
im = im.resize((int(ratio * d) for d in im.size), Image.BILINEAR)

data = get_data(im, model)
data1 = get_data(im, model1)

with torch.no_grad():
print(data.shape)
start_time = time.time()
scores, boxes, classes = model(data)
print("--- %s seconds ---" % (time.time() - start_time))
scores, boxes, classes = model1(data1)
print("--- %s seconds ---" % (time.time() - start_time))
print(scores, boxes, classes)
def get_data(im, model):
mean = [0.485, 0.456, 0.406]
std = [0.229, 0.224, 0.225]

data = torch.ByteTensor(torch.ByteStorage.from_buffer(im.tobytes()))
data = data.float().div(255).view(*im.size[::-1], len(im.mode))
data = data.permute(2, 0, 1)

for t, mean, std in zip(data, mean, std):
t.sub_(mean).div_(std)

pw, ph = ((model.stride - d % model.stride) % model.stride for d in im.size)
data = F.pad(data, (0, pw, 0, ph))
data = data.unsqueeze(0).cuda()
return data
infer_trt('asd.png', 256, 512)

**Comment by james-nvidia** [comment]
What's your overall objective? Are you independently inferring two models, or are you using the output of one to make a decision about the second model?
We normally recommend that you use `retinanet infer model1.pth` and `retinanet infer model2.pth` to infer your models. Is there a reason why you can't do that here?

**Comment by aidonchuk** [comment]
Hi! James thanks for reply. I'm using crop of output or like in my example i read another image from disk. 
I can't do that(retinanet infer than retinanet infer again) because performance loading model takes to long. I find that if create a new thread and create a new cuda context and make inference in thread everything is ok but it looks like overhead and unnecessary complexity. 

I need a very accurate predictions with as much performance as I can get due to objects as my input date goes at big speed. 

2 stage in my case works perfectly by performance because only some photos have objects of interest and this objects very complex. 
That why I share my benchmark of resnet34fpn 1 class vs 10 class - 0.02 vs 0.47( strange and big difference)

If you think that is not an issue pls give me some direction, how to make Engine.cpp(or something other file) produce models that can work like in my example.

**Comment by james-nvidia** [comment]
Are you using video streams or JPG images? If so, could you use DeepStream? It can easily call a secondary model only when the primary model detects certain classes. DeepStream has the big advantage of keeping everything in GPU memory, so inference speeds will be very fast.

**Comment by aidonchuk** [comment]
I'm working with Basler cameras and pypylon driver(https://github.com/basler/pypylon) it produce numpy, I have my own pipeline based on microservices for many devices(Jetson/Xavier) every device connected to it own camera(s) after every result calculated they send data to master service. Maybe I can try DeepStream but not now, my software already created.

**Comment by james-nvidia** [comment]
Your solution sounds interesting, but is beyond the scope of this repo. Please open an issue with a specific problem, giving steps to reproduce and the error produced.
